ledger-tool/src/main.rs

```diff
@@ -942,6 +948,97 @@ fn main() {
                 }
             }
         }
+        ("convert-accounts", Some(arg_matches)) => {
+            let output_directory = value_t_or_exit!(arg_matches, "output_directory", String);
+            let genesis_config = open_genesis_config(&ledger_path);
+            match load_bank_forks(
+                arg_matches,
+                &ledger_path,
+                &genesis_config,
+                ProcessOptions::default(),
+            ) {
+                Ok((mut bank_forks, _bank_forks_info, _leader_schedule_cache, _snapshot_hash)) => {
+                    let mut root_bank = Arc::try_unwrap(bank_forks.remove(bank_forks.root()))
+                        .unwrap_or_else(|_| panic!("More than one reference"));
+                    // Convert root_bank.EpochStakes
+                    for (_, stakes) in root_bank.epoch_stakes.iter_mut() {
+                        for (pubkey, (_, vote_account)) in stakes.vote_accounts.iter_mut() {
+                            VoteStateVersions::convert_from_raw(vote_account, &pubkey);
+                        }
+                    }
+
+                    let index = root_bank
+                        .rc
+                        .accounts
+                        .accounts_db
+                        .accounts_index
+                        .read()
+                        .unwrap();
+
+                    for (pubkey, slot_list) in index.account_maps.iter() {
+                        for (slot, _) in slot_list.read().unwrap().iter() {
+                            let ancestors = vec![(*slot, 1)].into_iter().collect();
+                            let mut account = root_bank
+                                .rc
+                                .accounts
+                                .load_slow(&ancestors, pubkey)
+                                .map(|(acc, _slot)| acc)
+                                .unwrap_or_else(|| {
+                                    panic!("Couldn't get account for pubkey {}", pubkey)
+                                });
+                            if account.owner == solana_vote_program::id() {
+                                VoteStateVersions::convert_from_raw(&mut account, pubkey);
+                                root_bank.store_account_convert(*slot, pubkey, &account);
```

Comment:
@sakridge @ryoqun, my main question is will this dumb approach of reading out/converting/re-storing these accounts work, especially on accounts from older slots?

Reply:
This might work; but I'm unsure what you want to achieve here. This way you can certainly could rewrite serialized vote accounts in snapshot. But this changes the binary (right?) then, account.hash and slot_hash and any future votes and PoH as well. This is a cascading effect. We can technically rewrite those, but we stop here: we can't rewrite third-party's signatures.

So, I think the ledger integrity is compromised. Could you explain why snapshot rewrite is needed nevertheless? :)

Reply:
We want to keep the existing TdS ledger state, so all TdSers don't need to restart from genesis.   The actual ledger history is ok to compromise on TdS right now 😬